### PR TITLE
new package: py-pytaridx

### PR DESCRIPTION
**Pull request by bhatiaharsh**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pytaridx/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = "https://github.com/LLNL/pytaridx"
+    git      = "git@github.com:LLNL/pytaridx.git"
+    url      = "https://github.com/LLNL/pytaridx/archive/refs/tags/v1.0.2.tar.gz"
+
+    maintainers = ['bhatiaharsh']
+
+    version('1.0.2', sha256='b1de4d5224b61f9c5aa1edf24b4ba0bfe1c5d2eb358c7a9b7914734ffb4efe8b')
+    version('master',  branch='master')
+
+    depends_on('python@3:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytaridx/package.py]: Is it true that Python 2 isn't supported? The setup.py seems to mention 2.7. Also, this should be `type=('build', 'run')` if you keep this.

File: var/spack/repos/builtin/packages/py-pytaridx/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from spack import *
+
+
+class PyPytaridx(PythonPackage):
+    """Python module/library for creating and maintaining a rapidly searchable
+       index for a tar-file. This allows "direct access" of members (files) in
+       the tar archive."""
+
+    homepage = "https://github.com/LLNL/pytaridx"
+    git      = "git@github.com:LLNL/pytaridx.git"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytaridx/package.py]:
```suggestion
    git      = "https://github.com/LLNL/pytaridx.git"
```
I've had a lot of issues with `git@github.com:` when SSH keys aren't configured correctly. Downloading over https should be safer.

File: var/spack/repos/builtin/packages/py-pytaridx/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPytaridx(PythonPackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytaridx/package.py:9]: Does this project have any relationship to https://pypi.org/project/taridx/?

Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/py-pytaridx/package.py:9]: interesting. not that one, but this one is the same: https://pypi.org/project/pytaridx.

File: var/spack/repos/builtin/packages/py-pytaridx/package.py (hunk 1 above)
Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/py-pytaridx/package.py]: hmm. i think we need to go back and change the setup. at some point Frank and Tomas ported to Python3, and there were some differences in how P2 and P3 handle bytestreams. I am hazy on details but i think they ended up just moving to P3 and not bothering with fixing for P2.

Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/py-pytaridx/package.py]: made the suggested change

File: var/spack/repos/builtin/packages/py-pytaridx/package.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    """Python module/library for creating and maintaining a rapidly searchable
+       index for a tar-file. This allows "direct access" of members (files) in
+       the tar archive."""
+
+    homepage = "https://github.com/LLNL/pytaridx"
+    git      = "https://github.com/LLNL/pytaridx.git"
+    url      = "https://github.com/LLNL/pytaridx/archive/refs/tags/v1.0.2.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytaridx/package.py]:
```suggestion
    pypi = "pytaridx/pytaridx-1.0.2.tar.gz"
```
See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github. You may also need to update the checksum.

File: var/spack/repos/builtin/packages/py-pytaridx/package.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+class PyPytaridx(PythonPackage):
+    """Python module/library for creating and maintaining a rapidly searchable
+       index for a tar-file. This allows "direct access" of members (files) in
+       the tar archive."""
+
+    homepage = "https://github.com/LLNL/pytaridx"
+    git      = "https://github.com/LLNL/pytaridx.git"
+    pypi = "pytaridx/pytaridx-1.0.2.tar.gz"
+    #url      = "https://github.com/LLNL/pytaridx/archive/refs/tags/v1.0.2.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytaridx/package.py]: These commented-out lines are causing the style tests to fail. You can just remove these.

File: var/spack/repos/builtin/packages/py-pytaridx/package.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    maintainers = ['bhatiaharsh']
+
+    # use pypi
+    version('1.0.2', sha26='702c42ade13ae8688a56a8edfcd7e0e7512a489a22796c6cfdbcef677010ee47')
+
+    # use git
+    #version('1.0.2', sha256='b1de4d5224b61f9c5aa1edf24b4ba0bfe1c5d2eb358c7a9b7914734ffb4efe8b')
+    #version('master',  branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytaridx/package.py]: You can still keep this version for people who want to use master

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**